### ISSUE TITLE
[Question] Does the update dialog show once in the application lifecycle or everytime I want?

### ISSUE THREAD

**Issue by NTA-trongpq** [larryaasen/upgrader#272]
Firstly, sorry because my English is not good.
I have a question: **Does the update dialog show once in the application lifecycle or everytime I want?**
Is there a limit to the number of impressions per day?
If it will show up everytime I want what function do I need to use?
@larryaasen Sorry for mentioning you.
Thank you!

**Comment by NTA-trongpq** [comment]
Help help.

**Comment by NTA-trongpq** [comment]
What did I miss? Why am I not responding?
Thank you everyone.

**Comment by dwzrlp** [comment]
For me it only shows once and I need to force quit the app and reopen it for it to show again.

**Comment by larryaasen** [comment]
Look at the README for this parameter: `durationUntilAlertAgain`.

**Comment by NTA-trongpq** [comment]
> Look at the README for this parameter: `durationUntilAlertAgain`.
My mean is how do I call check new version everytime I want? For example, the Upgrader().checkNewVersion() function.

**Comment by ChauCM** [comment]
yes, a function that programmatically checks the new version is very much needed, especially for a device that uses for a specific app that never close.

**Comment by larryaasen** [comment]
This might work with the existing code:
```
final upgrader = Upgrader(debugLogging: true);
await upgrader.initialize();
final shouldDisplay = upgrader.shouldDisplayUpgrade();
```
You could also call `checkVersion()` instead of `shouldDisplayUpgrade()` to display an alert if needed like this:
```
final upgrader = Upgrader(debugLogging: true);
await upgrader.initialize();
upgrader.checkVersion(context: context);
```

**Comment by larryaasen** [comment]
I have a fix for this issue now. Look at this version preview version: [8.0.0-alpha.1](https://pub.dev/packages/upgrader/versions/8.0.0-alpha.1)

**Comment by larryaasen** [comment]
The original fix was not working, so I made another fix and released it in this version: [8.0.0-alpha.2](https://pub.dev/packages/upgrader/versions/8.0.0-alpha.2)
